packages/core/admin/server/controllers/authentication.js

```diff
@@ -19,6 +19,12 @@ module.exports = {
       return passport.authenticate('local', { session: false }, (err, user, info) => {
         if (err) {
           strapi.eventHub.emit('admin.auth.error', { error: err, provider: 'local' });
+          // if this is a recognized error, allow it to bubble up to user
+          if (err.details?.code === 'LOGIN_NOT_ALLOWED') {
```

Comment:
should we store these error codes in constants?

Reply:
Yes, absolutely, but outside of the scope here. We need to overhaul all of our error handling, will probably come with v5